### PR TITLE
ci: update versions in the gh action

### PR DESCRIPTION
**Pull request by subhamkrai**
this commit update the versions in the
action. example minikube, python, commitlint
and others.

Signed-off-by: subhamkrai <srai@redhat.com>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

**Review by leseb — COMMENTED** [review]
When did we decided to go with `manusa/actions-setup-minikube@master`?

**Comment by subhamkrai** [comment]
> When did we decide to go with `manusa/actions-setup-minikube@master`?

I can revert that but with this, we don't need to update the version always. do we have concerns using the master version?

**Comment by leseb** [comment]
> > When did we decide to go with `manusa/actions-setup-minikube@master`?
> 
> I can revert that but with this, we don't need to update the version always. do we have concerns using the master version?

The action has been rather stable and I'd prefer to use a tagged version, it usually means more stable.

**Comment by subhamkrai** [comment]
> > > When did we decide to go with `manusa/actions-setup-minikube@master`?
> > 
> > 
> > I can revert that but with this, we don't need to update the version always. do we have concerns using the master version?
> 
> The action has been rather stable and I'd prefer to use a tagged version, it usually means more stable.

right, I have changed that to the latest version.

**Review by leseb — APPROVED** [review]

File: .github/workflows/commitlint.yml
Comment (travisn) [inline .github/workflows/commitlint.yml:23]: Is there a release we could use? Master typically is just less stable.

Comment (subhamkrai) [reply .github/workflows/commitlint.yml]: yes, we have

**Review by travisn — APPROVED** [review]

**Comment by subhamkrai** [comment]
please hold this till #8170  is ready to merge

**Comment by subhamkrai** [comment]
Also, I'm seeing the frequent failure of helm test so want 100% sure before we merge. Thanks

**Review by leseb — CHANGES_REQUESTED** [review]
holding for https://github.com/rook/rook/pull/8170

**Comment by travisn** [comment]
> holding for #8170

Do we need to wait for #8170? Not sure how I see it blocking this. If the tests are no more unstable with these version updates, seems like we could go ahead and merge.

**Comment by leseb** [comment]
> > holding for #8170
> 
> Do we need to wait for #8170? Not sure how I see it blocking this. If the tests are no more unstable with these version updates, seems like we could go ahead and merge.

Dunno I just went according to https://github.com/rook/rook/pull/8182#issuecomment-867349344

**Comment by travisn** [comment]
> please hold this till #8170 is ready to merge

@subhamkrai What is the need to hold it? Shall we go ahead and merge?

**Comment by travisn** [comment]
discussed offline with @subhamkrai looks like we're ready to merge since the helm tests are passing now

**Review by leseb — APPROVED** [review]

**Pull request merged**
